client/build_mac_host.sh

```diff
@@ -1,25 +1,83 @@
 # Automates build process for Platform codegen on macOS.
 
-set -e
+set -ex
+
+# Defaults
+BUILD_TYPE="debug"
+
+# Parse CLI options
+while getopts ":ailrw" options; do
+    case $options in
+        a) FLAG_ANDROID=true ;;
+        i) FLAG_IOS=true ;;
+        l) FLAG_LINUX=true ;;
+        r) BUILD_TYPE="release" ;;
+        w) WORKAROUND_ROSETTA_ISSUE=true ;;
+    esac
+done
+
+## BUILD FOR CURRENT HOST ##
 
 CLIENT_DIR=$(realpath $(dirname $0))
 PLATFORM_DIR=$(realpath $CLIENT_DIR/..)
 RSHARP_DIR="$PLATFORM_DIR/crates/rsharp"
-
+HOST_TARGET_TRIPLE=$(rustc -vV | sed -n 's|host: ||p')
 
 # Build Platform core and demo libraries.
 cd $PLATFORM_DIR
-cargo build --all
+cross build --target $HOST_TARGET_TRIPLE -p unity_states
```

Comment:
why do you cross build here, only to do it again on line 42? Perhaps you did it for `cpp_sharp`?

Also, why `cross` instead of `cargo`? isnt the `HOST_TARGET_TRIPLE` the host machine's target and therefore already what `cargo` is using?

Reply:
This should use `cargo` directly. Updating.